faiss/impl/HNSW.cpp

```diff
@@ -660,6 +662,11 @@ int search_from_candidates(
         }
     }
 
+    // Completed collection of result. Run post processor.
+    collector->finalize(nres, I);
+    // Collector completed its task. Release all resource of the collector.
+    collectorFactory->deleteCollector(collector);
```

Comment:
do we need this? as these are classes their object should get destructed once these objects go out of scope.

Reply:
It depends how resource is created. At least, the factory create collector with `new` which need to be released manually.

Reply:
objects created with `new` for sure gets deallocated. Its the pointers that create the issue. For pointers it is recommended to use unique_pointer.